compiler/rustc_expand/src/lib.rs

```diff
@@ -1,5 +1,7 @@
+#![feature(bool_to_option)]
 #![feature(crate_visibility_modifier)]
 #![feature(decl_macro)]
+#![feature(destructuring_assignment)]
```

Comment:
This feature is currently not used anywhere inside the compiler. Given that you are only using it at one single place in rustc_expand and that avoiding it would the life of for example mrustc easier at the expense of writing just a tiny bit less nice code, I would err towards not using it.

Reply:
To be clear if using it had a bigger benefit for rustc itself, I would be fine with using it. It is just that I don't think the benefit for rustc outweighs the cost for mrustc in this particular case.

Reply:
This is more or less https://github.com/rust-lang/compiler-team/issues/358 again.
I want to use compiler for dogfooding recently implemented unstable features, without looking at what third-party projects may think about it.
Destructuring assignment is a convenient feature, if it's removed from this PR then it may be useful in a next PR.
Similarly, https://github.com/rust-lang/rust/pull/82641 starts using `#![feature(extended_key_value_attributes)]`, should it abandon it too?
At which point are we allowed to use these features? I don't want mrustc to decide that.

Reply:
Note that `#![feature(extended_key_value_attributes)]` will probably be stabilized soon anyway.